service/frontend/service.go

```diff
@@ -345,10 +349,22 @@ func (s *Service) Start() {
 	s.operatorHandler.Start()
 	s.handler.Start()
 
-	listener := s.grpcListener
 	logger.Info("Starting to serve on frontend listener")
-	if err := s.server.Serve(listener); err != nil {
-		logger.Fatal("Failed to serve on frontend listener", tag.Error(err))
+
+	// Start the gRPC and HTTP server (if any) in the background
```

Comment:
I'm changing this in https://github.com/temporalio/temporal/pull/4584, so after that lands, you can just start two goroutines that check return value independently, and not have to do the channel here

Reply:
I see. If that lands first I will change when I conflict.

Reply:
that one landed

Reply:
Will fix